File: backend/src/main/java/com/madhacks/madmarket/repository/Listing.java
package com.madhacks.madmarket.repository;

import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;

import java.util.Date;

@NoArgsConstructor
@Getter
@Setter
public class Listing {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long listingId;
    private Long buyerId;
    private Long listerId;
    private Long categoryId;
    private String name;
    private String description;
    private String price;
    private Date timestamp;
}

File: frontend/mad-market/src/app/objects/listing.ts
export class Listing {
  constructor(obj : any) {
    this.listingId = obj.listingId;
    this.buyerId = obj.buyerId;
    this.userId = obj.userId;
    this.categoryId = obj.categoryId;
    this.name = obj.name;
    this.price = obj.price;
    this.keywords = obj.keywords;
    this.timestamp = obj.timestamp;
  }

  public listingId: number;
  public buyerId: number;
  public userId: number;
  public categoryId: number;
  public name: string;
  public price: number;
  public keywords: string;
  public timestamp: string;
}
